### ISSUE TITLE
IllegalStateException when validateOnCheckout+validatePeriodically false

### ISSUE THREAD

**Issue by Condor70** [apereo/cas#2263]
I'm still in the process of upgrading from CAS 3.5.2 to 5.0.1, but I keep running into problems.

Beans.newBlockingConnectionPool assigns an unconfigured SearchValidator as LDAP pool validator. This might work for AD and OpenLDAP, but not for the IBM Directory Server I am using.

Since there is no way to configure the validator without modifying CAS code I figured I'd disable it completely by configuring the LDAP pool with validateOnCheckout = false and validatePeriodically = false.

Unfortunately, Ldaptive throws an IllegalStateException in AbstractConnectionPool.initialize when a validator is configured, but no validation options are set.

org.apereo.cas.configuration.support.Beans needs a minor fix:

```
    public static ConnectionPool newBlockingConnectionPool(final AbstractLdapProperties l) {
        final DefaultConnectionFactory bindCf = newConnectionFactory(l);
        final PoolConfig pc = newPoolConfig(l);
        final BlockingConnectionPool cp = new BlockingConnectionPool(pc, bindCf);

        cp.setBlockWaitTime(newDuration(l.getBlockWaitTime()));
        cp.setPoolConfig(pc);

        final IdlePruneStrategy strategy = new IdlePruneStrategy();
        strategy.setIdleTime(newDuration(l.getIdleTime()));
        strategy.setPrunePeriod(newDuration(l.getPrunePeriod()));

        cp.setPruneStrategy(strategy);
//--->
        if (pc.isValidateOnCheckIn() || pc.isValidateOnCheckOut() || pc.isValidatePeriodically()) {
        	cp.setValidator(new SearchValidator());
        }
//<---
        cp.setFailFastInitialize(l.isFailFast());

        LOGGER.debug("Initializing ldap connection pool for {} and bindDn {}", l.getLdapUrl(), l.getBindDn());
        cp.initialize();
        return cp;
    }

```

Ps. If you have an idea on how to make the validator configurable I would prefer that over disabling it completely.

**Comment by mmoayyed** [comment]
You want to look at 5.1 SNAPSHOT for validator configuration. Port back changes via a PR to 5.0.2 if needed.

**Comment by Condor70** [comment]
I looked at the 5.1-SNAPSHOT, and it looks promising. Unfortunately, I just found out that our IBM Directory server doesn't allow anonymous searches or binds, so both SearchValidator and CompareValidator are not usable.

This leaves me with setting validateOnCheckOut and validatePeriodically to false, which is still not possible with the 5.1-SNAPSHOT.

Can I suggest adding a third ldap[].validation.type "none"?
